### PR TITLE
[SCB-1318]RSAProviderTokenManager#validatedToken expired resource recycling

### PR DESCRIPTION
**Pull request by wuliuqi666**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java [hunk 1]
```diff
@@ -50,7 +52,11 @@ public boolean valid(String token) {
         LOGGER.error("token is expired");
         return false;
       }
-      if (validatedToken.contains(rsaToken)) {
+      validatedToken = CacheBuilder.newBuilder()
```
Comment (liubao68) [inline handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java]: This logic seems totally wrong, basic tests should be taken.

Comment (wuliuqi666) [reply handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java]: under modify

Comment (wuliuqi666) [reply handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java]: ok

File: handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java [hunk 2]
```diff
@@ -50,16 +52,16 @@ public boolean valid(String token) {
...
       }
 
-      String sign = rsaToken.getSign();
-      String content = rsaToken.plainToken();
-      String publicKey = getPublicKey(rsaToken.getInstanceId(), rsaToken.getServiceId());
-      boolean verify = RSAUtils.verify(publicKey, sign, content);
-      if (verify && !tokenExprired(rsaToken)) {
-        validatedToken.add(rsaToken);
+      if (isVerify(rsaToken) && !tokenExprired(rsaToken)) {
+        validatedToken.put(rsaToken, true);
```
Comment (liubao68) [inline handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java]: I think this patch does not fix the problem mentiond in this first commit. Each call to "valid", validatedToken is always empty(newly created).

File: handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java [hunk 3]
```diff
@@ -69,6 +71,18 @@ public boolean valid(String token) {
     }
   }
 
+  public boolean isVerify(RSAAuthenticationToken rsaToken)
```
Comment (liubao68) [inline handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java]: isValidToken maybe is better

**Comment by liubao68** [comment]
Any updates on this issue?

File: handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java (hunk 3 above)
Comment (wuliuqi666) [reply handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java]: under revision

**Pull request merged**